### PR TITLE
feat: S2I strategy for on cluster build

### LINKED ISSUE #989: On-cluster build: s2i support

### PR DESCRIPTION
**Pull request by zroubalik**
Signed-off-by: Zbynek Roubalik <zroubalik@gmail.com>

# Changes

- :gift: S2I strategy for on cluster build

The S2I Tekton Task that's been used by this feature needs changes, PR opened: https://github.com/tektoncd/catalog/pull/1025

/kind enhancement

Fixes #989

### CODE REVIEW

File: pipelines/tekton/validate.go [hunk 1]
```diff
@@ -23,16 +24,23 @@ func (e ErrRuntimeNotSupported) Error() string {
...
-	}
-
-	if len(f.Buildpacks) > 0 {
-		return ErrBuilpacksNotSupported
+	if f.Builder == fn.BuilderPack {
+		if f.Runtime == "" {
+			return ErrRuntimeRequired
+		}
+
+		if f.Runtime == "go" || f.Runtime == "rust" {
```
Comment (lance) [inline pipelines/tekton/validate.go:32]: What about a check here for Python and SpringBoot, and test cases below?

**Review by lance — COMMENTED** [review]
Just a small question re: springboot and python

File: pipelines/tekton/validate.go (hunk 1 above)
Comment (zroubalik) [reply pipelines/tekton/validate.go:32]: I think that both are functional, @jrangelramos is that right?

Comment (jrangelramos) [reply pipelines/tekton/validate.go:32]: Yes @zroubalik Python and Springboot are functional

**Comment by lance** [comment]
https://github.com/tektoncd/catalog/pull/1025 has landed
/unhold

**Comment by lance** [comment]
@zroubalik this is failing to merge because of a > 10% difference in code coverage. Can you do anything to improve this so the PR will land?

**Review by lkingland — APPROVED** [review]
/lgtm

**Comment by zroubalik** [comment]
the dependent release have been moved to 22nd August :(

**Comment by lance** [comment]
@zroubalik do you want to hold for 4 more days then?

**Comment by zroubalik** [comment]
@lance yes, I won't to be sure that it is fully functional before merging

**Comment by zroubalik** [comment]
Closing in favor of: https://github.com/knative-sandbox/kn-plugin-func/pull/1191

**Pull request closed without merging**